### PR TITLE
Adding initial work with 100% test coverage and API docs.

### PR DESCRIPTION
**Pull request by GoogilyBoogily**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+[![Build Status][travis-badge]][travis-link]
+[![Dependencies][dependencies-badge]][dependencies-link]
+[![Dev Dependencies][devdependencies-badge]][devdependencies-link]
+[![codecov.io][codecov-badge]][codecov-link]
+
+
+Overview
+--------
+
+Simplifies the process of adding all schemas from a folder or a single file. This [tv4][https://www.npmjs.com/package/tv4] module as the underlying schema validator. Because of this, some of tv4's functions are exposed.
```
Comment (quantumew) [inline README.md]: I am guessing you meant to make that a link?
```
[tv4](https://www.npmjs.com/package/tv4)
```

File: lib/index.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+/**
+ * Describes problems encountered during validation.
+ *
+ * @typedef {Object} validationProblems
+ * @property {boolean} valid Always set to false.
+ * @property {Array.<Object>} errors
+ * @property {Array.<string>} missing
+ */
+
+/**
```
Comment (quantumew) [inline lib/index.js]: Two newlines around JSDoc blocks.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    "jasmine": "~2.5.3",
+    "jasmine-test-helpers": "~1.1.2"
+  },
+  "dependencies": {
+    "bluebird": "^3.4.7",
+    "glob": "^7.1.0",
+    "mock-require": "^2.0.1",
+    "tv4": "^1.2.7"
+  }
+}
```
Comment (quantumew) [inline package.json]: Do we care that this is only indented two spaces? I know I forget this with package.json files.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
+{
+  "name": "schema-validator",
+  "version": "1.0.0",
+  "description": "Manages loading schemas from a single file or a directory and validation for checking data against those schemas.",
+  "main": "lib/index.js",
+  "scripts": {
+    "test": "script/test",
+    "watch": "script/watch"
```
Comment (quantumew) [inline package.json]: These scripts don't exist

File: spec/lib/index.spec.js [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+    var mock, path, promiseMock, SchemaValidator, schemaValidatorInstance;
+
+    beforeEach(() => {
+        var fs, tv4;
+
+        fs = require("fs");
+        mock = require("mock-require");
+        path = require("path");
+        spyOn(path, "resolve").and.callThrough();
+        promiseMock = require("../mock/promise-mock")();
```
Comment (quantumew) [inline spec/lib/index.spec.js]: Why are we mocking out promises?

File: spec/lib/index.spec.js [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+            expect(pattern).toBe("./folder/**/*.json");
+            expect(options).toEqual({
+                strict: true,
+                nodir: true
+            });
+
+            return promiseMock.resolve([
+                "/folder/email.json",
+                "/folder/folder/number.json"
+            ]);
```
Comment (quantumew) [inline spec/lib/index.spec.js]: You can just use `bluebird.resolve` here, right?

File: spec/mock/promise-mock.js [hunk 7]
```diff
@@ -0,0 +1,324 @@
+"use strict";
+
+/* global Promise */
+module.exports = () => {
```
Comment (quantumew) [inline spec/mock/promise-mock.js]: I think you can just remove this mock. I don't see a purpose for it

**Review by quantumew — CHANGES_REQUESTED** [review]

File: spec/mock/promise-mock.js (hunk 7 above)
Comment (GoogilyBoogily) [reply spec/mock/promise-mock.js]: Removed. ac79550

File: spec/lib/index.spec.js (hunk 6 above)
Comment (GoogilyBoogily) [reply spec/lib/index.spec.js]: I tested it, and yeah you can. Removed promiseMock. ac79550

File: spec/lib/index.spec.js (hunk 5 above)
Comment (GoogilyBoogily) [reply spec/lib/index.spec.js]: I am actually unsure. I was following OpenToken's example. Removing it though. ac79550

File: package.json (hunk 3 above)
Comment (GoogilyBoogily) [reply package.json]: I think we do :/ Fixed ac79550.

File: package.json (hunk 4 above)
Comment (GoogilyBoogily) [reply package.json]: Crap, they must have disappeared when I was fixing the commits. Fixed ac79550.

File: lib/index.js (hunk 2 above)
Comment (GoogilyBoogily) [reply lib/index.js]: Fixed ac79550.

File: README.md (hunk 1 above)
Comment (GoogilyBoogily) [reply README.md]: Crap, yeah. Fixed ac79550.

File: package.json (hunk 3 above)
Comment (quantumew) [reply package.json]: I don't see this fixed

**Review by quantumew — APPROVED** [review]
Looks like the package file didn't get updated but other then that this looks good.

**Pull request merged**